src/shogun/classifier/LDA.cpp

```diff
@@ -150,12 +140,14 @@ template <typename ST>
 bool CLDA::solver_classic()
 {
 	auto dense_feat = static_cast<CDenseFeatures<ST>*>(features);
-	index_t num_feat = dense_feat->get_num_features();
+  auto labels = multiclass_labels(m_labels);
+  REQUIRE(labels->get_num_classes()==2, "Number of classes exceeds 2 for binary classifier: %d provided\n", labels->get_num_classes())
```

Comment:
cant this check be pulled out and only done once?

Reply:
i am not sure how.... I tried doing it at LDASolver but that messed up FisherLDA so i guess its better to do it here.